src-tauri/src/main.rs

```diff
@@ -72,12 +81,59 @@ async fn control_grabbing(
     Ok(())
 }
 
+/// Engages the logger.
+fn engage_logger() -> Result<log4rs::Handle, SetLoggerError> {
+    let level: log::LevelFilter = option_env!("RUST_LOG")
+        .and_then(|s| log::LevelFilter::from_str(s).ok())
+        .unwrap_or(log::LevelFilter::Warn);
+
+    let file_path = wooting_macro_backend::config::LogFilePath::file_name();
+
+    // Build a stderr logger.
+    let stderr = ConsoleAppender::builder()
+        .encoder(Box::new(PatternEncoder::new(
+            "{d(%Y-%m-%d %H:%M:%S:%f)} | {h({({l}):5.5})} | {m}{n}",
+        )))
+        .target(Target::Stderr)
+        .build();
+
+    // Logging to log file.
+    let logfile = FileAppender::builder()
+        // Pattern: https://docs.rs/log4rs/*/log4rs/encode/pattern/index.html
+        .encoder(Box::new(PatternEncoder::new(
+            "{d(%Y-%m-%d %H:%M:%S:%f)} | {h({({l}):5.5})} | {m}{n}",
+        )))
+        .build(file_path)
+        .unwrap();
+
+    // Log Trace level output to file where trace is the default level
+    // and the programmatically specified level to stderr.
+    let config = Config::builder()
+        .appender(Appender::builder().build("logfile", Box::new(logfile)))
+        .appender(
+            Appender::builder()
+                .filter(Box::new(ThresholdFilter::new(level)))
+                .build("stderr", Box::new(stderr)),
+        )
+        .build(
+            Root::builder()
+                .appender("logfile")
+                .appender("stderr")
+                .build(level),
+        )
+        .unwrap();
+
+    let handle = log4rs::init_config(config)?;
+    Ok(handle)
+}
+
 #[tokio::main(flavor = "multi_thread", worker_threads = 10)]
 /// Spawn the backend thread.
 /// Note: this doesn't work on macOS since we cannot give the thread the proper permissions
 /// (will crash on key grab/listen)
 async fn main() {
-    env_logger::init();
+    //This is here for possible future logging configuration from the frontend
+    let _log_config_var = engage_logger().unwrap();
```

Comment:
Not really much of a need to declare a variable here

Additionally, it would be better to do `.expect("Failed to engage logger")` rather than `.unwarp()` as it has the same behaviour, but in the case that it occurs, it gives more context

Reply:
the idea was to use this a little bit later (next patch even) to allow setting the logging on the fly using config values. But i can take it out

Reply:
Fixed in eb8c9a6